content/client-lib-development-guide/features.textile

```diff
@@ -1281,15 +1281,50 @@ h4. Stats
+* @(TS3)@ This clause has been deleted.
+* @(TS4)@ @Stats.ConnectionTypes@ - contains a breakdown of summary stats data for different (TLS vs non-TLS) connection types. Has the following attributes:
+** @(TS4a)@ @tls@ - a @ResourceCount@. The number of TLS connections.
+** @(TS4b)@ @plain@ - a @ResourceCount@. The number of non-TLS connections.
+** @(TS4c)@ @all@ - a @ResourceCount@. The number of all connections (both TLS & non-TLS).
+* @(TS5)@ @Stats.MessageCount@ - contains aggregate counts for messages and data transferred. Has the following attributes:
+** @(TS5a)@ @count@ - integer. The count of all messages.
+** @(TS5b)@ @data@ - integer. The total number of bytes transferred for all messages.
+* @(TS6)@ @Stats.MessageTypes@ - contains a breakdown of summary stats data for different (channel vs presence) message types. Has the following attributes:
+** @(TS6a)@ @messages@ - a @MessageCount@. The count of channel messages.
+** @(TS6b)@ @presence@ - a @MessageCount@. The count of presence messages.
+** @(TS6c)@ @all@ - a @MessageCount@. The count of all messages (channel & presence).
+* @(TS7)@ @Stats.MessageTraffic@ - contains a breakdown of summary stats data for traffic over various transport types. Has the following attributes:
+** @(TS7a)@ @realtime@ - a @MessageTypes@. The count of messages transferred over a realtime transport such as WebSocket.
+** @(TS7b)@ @rest@ - a @MessageTypes@. The count of messages transferred over a realtime transport such as WebSocket.
+** @(TS7c)@ @webhook@ - a @MessageTypes@. The count of messages delivered using webhooks.
+** @(TS7d)@ @all@ - a @MessageTypes@. The count of all messages (includes realtime, rest and webhook messages).
+* @(TS8)@ @Stats.RequestCount@ - contains aggregate counts for requests made. Has the following attributes:
+** @(TS8a)@ @succeeded@ - integer. The number of requests that succeeded.
+** @(TS8b)@ @failed@ - integer. The number of requests that failed.
+** @(TS8c)@ @refused@ - integer. The number of requests that were refused, typically as a result of permissions or a limit being exceeded.
+* @(TS9)@ @Stats.ResourceCount@ - contains aggregate data for usage of a resource in a specific scope. Has the following attributes:
+** @(TS9a)@ @opened@ - integer. The total number of resources opened of this type.
+** @(TS9b)@ @peak@ - integer. The peak number of resources of this type used for this period.
+** @(TS9c)@ @mean@ - float. The average number of resources of this type used for this period.
+** @(TS9d)@ @min@ - integer. The minimum total resources of this type used for this period.
+** @(TS9e)@ @refused@ - integer. The number of resource requests refused within this period.
+* @(TS10)@ @Stats.PushStats@ - detailed stats on push notifications. Has the following attributes:
+** @(TS10a)@ @messages@ - integer. Total number of Push channel messages.
```

Comment:
Going purely on the wikipedia page for 'push technology' I'd say we should use lowercase 'p' for 'push'